packages/wagmi/src/client.ts

```diff
@@ -449,8 +502,33 @@ export class Web3Modal extends Web3ModalScaffold {
         }
       })
 
-      provider.onRpcResponse(() => {
-        super.close()
+      provider.onRpcResponse(receive => {
+        // eslint-disable-next-line @typescript-eslint/prefer-ts-expect-error, @typescript-eslint/ban-ts-comment
+        // @ts-ignore
+        const payload = receive?.payload
+        // eslint-disable-next-line @typescript-eslint/prefer-ts-expect-error, @typescript-eslint/ban-ts-comment
+        // @ts-ignore
+        const isError = receive?.type === '@w3m-frame/RPC_REQUEST_ERROR'
+
+        if (isError && super.isOpen()) {
+          if (super.isTransactionStackEmpty()) {
+            super.close()
+          } else {
+            super.popTransactionStack(true)
+          }
+        }
+
+        const isPayloadString = typeof payload === 'string'
+        const isAddress = isPayloadString ? payload?.startsWith('0x') : false
+        const isCompleted = isAddress && payload?.length > 10
```

Comment:
what happens with responses that are not 0x strings?

Reply:
+1 on this. This feels a bit hacky and unreliable.

Reply:
@svenvoskamp @tomiir I think we could make this by checking the string if it's transaction has or not: https://github.com/WalletConnect/web3modal/pull/2116/commits/5f6f129610911ed4b2d1b24709fe63839aff9639